Source/Core/Core/HW/WiimoteEmu/MotionPlus.cpp

```diff
@@ -17,47 +21,63 @@ MotionPlus::MotionPlus() : Extension("MotionPlus")
 
 void MotionPlus::Reset()
 {
-  reg_data = {};
+  m_reg_data = {};
 
-  constexpr std::array<u8, 6> initial_id = {0x00, 0x00, 0xA6, 0x20, 0x00, 0x05};
+  m_activation_progress = {};
 
-  // FYI: This ID changes on activation
-  std::copy(std::begin(initial_id), std::end(initial_id), reg_data.ext_identifier);
+  // FYI: This ID changes on activation/deactivation
+  constexpr std::array<u8, 6> initial_id = {0x00, 0x00, 0xA6, 0x20, 0x00, 0x05};
+  m_reg_data.ext_identifier = initial_id;
 
-  // TODO: determine meaning of calibration data:
-  constexpr std::array<u8, 32> cdata = {
+  // Calibration data.
+  // Copied from real hardware as it has yet to be fully reverse engineered.
+  // It's possible a checksum is present as the other extensions have one.
+  constexpr std::array<u8, 32> cal_data = {
       0x78, 0xd9, 0x78, 0x38, 0x77, 0x9d, 0x2f, 0x0c, 0xcf, 0xf0, 0x31,
       0xad, 0xc8, 0x0b, 0x5e, 0x39, 0x6f, 0x81, 0x7b, 0x89, 0x78, 0x51,
       0x33, 0x60, 0xc9, 0xf5, 0x37, 0xc1, 0x2d, 0xe9, 0x15, 0x8d,
+      // 0x79, 0xbc, 0x77, 0xa3, 0x76, 0xd9, 0x30, 0x6c, 0xce, 0x8a, 0x2b,
+      // 0x83, 0xc8, 0x02, 0x0e, 0x70, 0x74, 0xb5, 0x79, 0x8e, 0x76, 0x45,
+      // 0x38, 0x22, 0xc7, 0xd6, 0x32, 0x3b, 0x2d, 0x35, 0xde, 0x37,
   };
-
-  std::copy(std::begin(cdata), std::end(cdata), reg_data.calibration_data);
-
-  // TODO: determine the meaning behind this:
-  constexpr std::array<u8, 64> cert = {
-      0x99, 0x1a, 0x07, 0x1b, 0x97, 0xf1, 0x11, 0x78, 0x0c, 0x42, 0x2b, 0x68, 0xdf,
-      0x44, 0x38, 0x0d, 0x2b, 0x7e, 0xd6, 0x84, 0x84, 0x58, 0x65, 0xc9, 0xf2, 0x95,
-      0xd9, 0xaf, 0xb6, 0xc4, 0x87, 0xd5, 0x18, 0xdb, 0x67, 0x3a, 0xc0, 0x71, 0xec,
-      0x3e, 0xf4, 0xe6, 0x7e, 0x35, 0xa3, 0x29, 0xf8, 0x1f, 0xc5, 0x7c, 0x3d, 0xb9,
-      0x56, 0x22, 0x95, 0x98, 0x8f, 0xfb, 0x66, 0x3e, 0x9a, 0xdd, 0xeb, 0x7e,
-  };
-
-  std::copy(std::begin(cert), std::end(cert), reg_data.cert_data);
+  // constexpr std::array<u8, 32> cal_data = {
+  //     0x7d, 0xe2, 0x80, 0x5f, 0x78, 0x56, 0x31, 0x04, 0xce, 0xce, 0x33,
+  //     0xf9, 0xc8, 0x04, 0x63, 0x22, 0x77, 0x26, 0x7c, 0xb7, 0x79, 0x62,
+  //     0x34, 0x56, 0xc9, 0xa3, 0x3a, 0x35, 0x2d, 0xa8, 0xa9, 0xbc,
```

Comment:
Apart from that `0xc8` in the middle and the `0x2d` in the end I don't really see anything that feels static; and the majority looks like some sort of pattern, but nothing ends up as useful values when packing them into floats or whatever (the only thing I didn't try is doubles; but I kinda doubt they're using those?)
We should _really_ grab that part of the IOS and see what we can squeeze out of it.